test/toniq/job_process_test.exs

```diff
@@ -25,20 +25,22 @@ defmodule Toniq.JobProcessTest do
     assert Toniq.JobProcess.run(job) == {:job_was_successful, job}
   end
 
-  test "a job that raises an error returns {:job_has_failed, job, error}" do
+  test "a job that raises an error returns {:job_has_failed, job, error, stack}" do
     job = %{worker: TestErrorWorker, arguments: [data: 10]}
-    assert Toniq.JobProcess.run(job) == {:job_has_failed, job, %RuntimeError{message: "fail"}}
+    result = Toniq.JobProcess.run(job)
+    assert {:job_has_failed, ^job, %RuntimeError{message: "fail"}, _} = result
+    assert {_, _, _, [{TestErrorWorker, :perform, 1, _}|_]} = result
```

Comment:
Looks, good. A couple of things though before I merge this.

I've written the other tests using "assert actual == expected", so I'd prefer if these where like that too. If for nothing else then for consistency.

Another reason to use actual == expected is that then ExUnit can give us better error messages.

When using `expected = actual` and relying on pattern matching you get:

![](https://s3-eu-west-1.amazonaws.com/uploads-eu.hipchat.com/10794/29896/Ow0TChADf2XJ2AJ/Screen%20Shot%202016-09-17%20at%2011.22.52.png)

When doing`actual == expected` you get:

![](https://s3-eu-west-1.amazonaws.com/uploads-eu.hipchat.com/10794/29896/n5OTiaN9puqV4Vf/Screen%20Shot%202016-09-17%20at%2011.23.39.png)

Also, do you think we need to assert something about the stack trace? This test would pass even if it was "nil".